Added the Table of Contents

wonntann opened the pull request:
Linked to the github blog post for the png image to visualize the generated table of contents from the headings

<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [types-of-contributions.md](/main/contributing/types-of-contributions.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why:

Closes [issue link]

<!--
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed:

<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. If you made changes to the `content` directory, a table will populate in a comment below with the staging and live article links -->

### Check off the following:

- [ ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

<!-- Description of the writer impact here -->

wonntann committed:
Added the Table of Contents 

Linked to the github blog post for the png image to visualize the generated table of contents from the headings

ramyaparimi commented:
@wonntann 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

felicitymay commented on content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md:
```diff
@@ -24,6 +24,11 @@ To create a heading, add one to six <kbd>#</kbd> symbols before your heading tex
 
+The use of two or more headings automatically generates a table of contents within the header. Each respective heading title will be listed in the table of contents and navigate within the document to the selected section. 
+
+![Headings Table of Contents](https://i2.wp.com/user-images.githubusercontent.com/7900087/113821370-df915480-9730-11eb-8aed-bdc50e2212d5.gif?ssl=1)
```
This image isn't showing up in the [preview](https://docs-15218--patch-2.herokuapp.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#headings). We're also trying stop avoid using animated GIFs in the help as they are bad for accessibility. 

Can I suggest that you add the image below to the `images` directory [here](https://github.com/wonntann/docs/upload/patch-2/assets/images/help/repository) as `table-of-contents.png`? Then we can update this to:

```suggestion
![Screenshot highlighting the table of contents icon](/assets/images/help/repository/table-of-contents.png)
```

<img width="889" alt="table-of-contents" src="https://user-images.githubusercontent.com/1877141/153183249-67e2811d-6d48-4647-b3cc-37bde91c5b4f.png">

wonntann replied:
Ok, sounds good. I committed a png image for the toc. I overlooked the accessibility, if you need any assistance with increasing accessibility please let me know. Thanks!

felicitymay reviewed (commented):
@wonntann - thank you so much for persevering and creating a new PR to replace https://github.com/github/docs/pull/13297 💖 

This looks good. I've made one suggestion to bring the text more into line with our current guidelines which I'll merge, and a second suggestion to fix the image.

When you've added the suggested image, we should be ready to merge this PR and publish your changes 🎉

felicitymay committed:
Update content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md

wonntann committed:
image of headings table of contents

image displaying the table of contents featuring the headings in the respective document

felicitymay committed:
Update content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md

felicitymay committed:
Update content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md

felicitymay commented on content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md:
```diff
@@ -24,6 +24,11 @@ To create a heading, add one to six <kbd>#</kbd> symbols before your heading tex
 
 ![Rendered H1, H2, and H6 headings](/assets/images/help/writing/headings-rendered.png)
 
+When you use of two or more headings, GitHub automatically generates a table of contents which you can access by clicking {% octicon "list-unordered" aria-label="The unordered list icon" %} within the file header. Each heading title is listed in the table of contents and you can click a title to navigate to the selected section. 
```
Fix a typo that I introduced 🤦🏻‍♀️ 
```suggestion
When you use two or more headings, GitHub automatically generates a table of contents which you can access by clicking {% octicon "list-unordered" aria-label="The unordered list icon" %} within the file header. Each heading title is listed in the table of contents and you can click a title to navigate to the selected section. 
```

felicitymay committed:
Update content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md

4809381693 commented:
> Linked to the github blog post for the png image to visualize the generated table of contents from the headings
> 
> <!--
> Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.
> 
> See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.
> 
> For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).
> 
> We cannot accept changes to our translated content right now. See the [types-of-contributions.md](/main/contributing/types-of-contributions.md#earth_asia-translations) for more information.
> 
> Thanks again!
> -->
> 
> ### Why:
> 
> Closes [issue link]
> 
> <!--
> - If there's an existing issue for your change, please link to it.
> - If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->
> 
> ### What's being changed:
> 
> <!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. If you made changes to the `content` directory, a table will populate in a comment below with the staging and live article links -->
> 
> ### Check off the following:
> 
> - [ ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
> - [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).
> 
> ### Writer impact (This section is for GitHub staff members only):
> 
> - [ ] This pull request impacts the contribution experience
>   - [ ] I have added the 'writer impact' label
>   - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")
> 
> <!-- Description of the writer impact here -->
>

Chipobg commented on content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md:
```diff
@@ -24,6 +24,11 @@ To create a heading, add one to six <kbd>#</kbd> symbols before your heading tex
 
 ![Rendered H1, H2, and H6 headings](/assets/images/help/writing/headings-rendered.png)
 
+The use of two or more headings automatically generates a table of contents within the header. Each respective heading title will be listed in the table of contents and navigate within the document to the selected section. 
```
[]()

wonntann committed:
Merge branch 'main' into patch-2

wonntann commented:
Sounds great. I have made the edits for the image. Do you mind guiding me on how to update the image without having to do a different pull request?

wonntann committed:
Merge branch 'github:main' into patch-2

wonntann commented:
>   Thanks @wonntann - happy to give a try.


Thanks for the walk through, I am not able to do either of those options, since I didn't work on it locally nor can I clone it at the moment.

I original added the image via the "Make a contribution" link on the github docs page for this topic at https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

felicitymay commented:
@wonntann - One last try, I've just created a test PR from a fork so that I ought to be able to give you better instructions.

1. Click on your branch name at the top of the page `wonntann:patch-2`. This should take you to: https://github.com/wonntann/docs/tree/patch-2.
2. Navigate to the folder that contains the image and click the image (I think this might be: https://github.com/wonntann/docs/blob/patch-2/assets/images/help/repository/headers_toc.png).
3. At the top right of the header for the file, you should see a bin icon. Click this to delete the current image.
4. Commit that change.
5. On the next page, you should find yourself in the `/assets/images/help/repository/` with an **Add file** button. You ought to be able to add the cropped image to the folder and commit this change.

If this doesn't work, I'll find an alternative solution. Thanks for your patience 💖

wonntann committed:
Delete headers_toc.png

wonntann committed:
Replacing image

Updating the headers_toc.png image as per https://github.com/github/docs/pull/15218#issuecomment-1035496615

wonntann commented:
> If this doesn't work, I'll find an alternative solution. Thanks for your patience sparkling_heart
Thank you so much @felicitymay 
Please let me know if there needs to be any changes on my end. Have a great day!

felicitymay commented on content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md:
```diff
@@ -24,6 +24,11 @@ To create a heading, add one to six <kbd>#</kbd> symbols before your heading tex
 
+When you use two or more headings, GitHub automatically generates a table of contents which you can access by clicking {% octicon "list-unordered" aria-label="The unordered list icon" %} within the file header. Each heading title is listed in the table of contents and you can click a title to navigate to the selected section. 
+
+![Screenshot highlighting the table of contents icon](/assets/images/help/repository/headers_toc.png)
```
```suggestion
![Screenshot highlighting the table of contents icon](/assets/images/help/repository/headings_toc.png)
```

felicitymay committed:
Update content/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.md

felicitymay committed:
Merge branch 'main' into patch-2

felicitymay commented:
> Please let me know if there needs to be any changes on my end. Have a great day!

@wonntann - thanks for persevering and fixing the image 💖 

I think this PR is ready to merge, but I'd like to check the preview first, just to confirm that all is well. Unfortunately, there seems to be a problem with the preview. I'm checking with our engineers about this and hope to be able to merge this PR shortly. After that, your changes will be live within around 24 hours.

felicitymay committed:
Merge branch 'main' into patch-2

felicitymay committed:
Merge branch 'main' into patch-2

felicitymay commented:
Thanks to our wonderful engineers the previews are now working correctly and we can merge this PR 🚀 🥳 

It may take up to 24 hours before you see your change in the live docs. Thanks for your contribution @wonntann 💖
